source/extensions/filters/http/oauth2/filter.cc

```diff
@@ -93,6 +93,26 @@ authScopesList(const Protobuf::RepeatedPtrField<std::string>& auth_scopes_protos
   return scopes;
 }
 
+// Transforms the proto list into encoded resource params
+// Takes care of percentage encoding http and https is needed
+std::string encodeResourceList(const Protobuf::RepeatedPtrField<std::string>& resources_protos) {
+  std::string result = "";
+
+  if (!resources_protos.empty()) {
```

Comment:
no need for this since in the empty case the for loop would terminate immediately